scanpy/preprocessing/_pca.py

```diff
@@ -142,52 +141,26 @@ def pca(
         adata = AnnData(data, dtype=data.dtype)
 
     # Check for use_highly_varible
-    if use_highly_variable is True and 'highly_variable' not in adata.var.keys():
+    if use_highly_variable is not None:
         warn(
             "Argument `use_highly_variable` is deprecated, consider using a mask argument.",
             FutureWarning,
         )
+    if use_highly_variable and mask is not None:
         raise ValueError(
-            'Did not find adata.var[\'highly_variable\']. '
-            'Either your data already only consists of highly-variable genes '
-            'or consider running `pp.highly_variable_genes` first.'
+            'These arguments are incompatible.'
+            'Use_highly_variable can be called through mask="highly_variable"'
         )
+    elif use_highly_variable or (use_highly_variable is not None):
```

Comment:
This check simplifies to `use_highly_variable is not None`, and would return `True` if `use_highly_variable=False`